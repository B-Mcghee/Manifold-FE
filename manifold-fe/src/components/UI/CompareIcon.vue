<template>
  <svg
    id="Layer_1"
    data-name="Layer 1"
    xmlns="http://www.w3.org/2000/svg"
    viewBox="0 0 109.3 106.45"
    :height="height"
    :width="width"
    @click="compareMeteorite"
  >
    <transition name="border">
      <rect
        id="border"
        v-if="animate"
        x="1"
        y="1"
        width="107"
        height="104"
        stroke="#000"
        stroke-miterlimit="10"
        :stroke-width="stroke"
        :fill="border"
      />
    </transition>

    <path
      d="M102.83,103.87a3.54,3.54,0,0,1-2.47-1L68.54,71.7a3.62,3.62,0,0,1-.07-5.09,3.51,3.51,0,0,1,5-.07L105.3,97.71a3.62,3.62,0,0,1,.07,5.08A3.49,3.49,0,0,1,102.83,103.87Z"
      transform="translate(0.92 0.58)"
    />
    <path
      d="M71,67.53a1.5,1.5,0,0,1,1.06.44L103.9,99.14a1.61,1.61,0,0,1,0,2.26,1.52,1.52,0,0,1-2.17,0L69.94,70.27a1.61,1.61,0,0,1,0-2.27,1.5,1.5,0,0,1,1.1-.47m0-4a5.52,5.52,0,0,0-4,1.69h0a5.61,5.61,0,0,0,.11,7.9L99,104.29a5.53,5.53,0,0,0,7.85-.11h0a5.63,5.63,0,0,0-.11-7.9L74.87,65.11A5.53,5.53,0,0,0,71,63.53Z"
      transform="translate(0.92 0.58)"
    />
    <path
      d="M3,40.59c.84,50.11,74.87,50.12,75.72,0C77.88-9.52,3.85-9.53,3,40.59Z"
      transform="translate(0.92 0.58)"
      fill="#c9fbff"
      stroke="#000"
      stroke-miterlimit="10"
      stroke-width="6"
    />
  </svg>
</template>

<script>
export default {
  data: () => ({
    height: "1em",
    width: "1em",
    fill: "#fff",
    stroke: 2,
    animate: false,
  }),
  methods: {
    compareMeteorite() {
      this.animate = !this.animate;

      this.fill == "#fff" ? (this.fill = "#1b579e") : (this.fill = "#fff");
    },
  },
  computed: {
    border() {
      const fill = this.fill;
      return fill;
    },
  },
};
</script>

<style>
#border {
  animation-name: bookmark-complete;
  animation-duration: 1s;
  animation-iteration-count: 1;
}
.border-enter {
  opacity: 0;
  stroke-dasharray: 0;
}

.border-enter-active,
.border-leave-active {
  transition: all 0.5s ease-in-out;
}

.border-enter-from,
.border-leave-to {
  opacity: 0;
  stroke-dasharray: 0;
}
</style>
