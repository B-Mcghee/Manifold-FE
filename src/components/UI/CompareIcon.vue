<template>
  <v-tooltip bottom>
    <template v-slot:activator="{ on, attrs }">
      <v-btn
        width="100%"
        @click.prevent="compareMeteorite"
        v-bind="attrs"
        v-on="on"
      >
        <svg
          id="Layer_1"
          data-name="Layer 1"
          xmlns="http://www.w3.org/2000/svg"
          viewBox="0 0 163 158.49"
          :height="height"
          :width="width"
        >
          <transition v-if="compare" name="border">
            <rect
              x="1"
              y="1"
              width="161"
              height="156.49"
              stroke="#000"
              stroke-miterlimit="10"
              :stroke-width="stroke"
              fill="#1b579e"
            />
          </transition>
          <transition v-else name="border">
            <rect
              x="1"
              y="1"
              width="161"
              height="156.49"
              stroke="#000"
              stroke-miterlimit="10"
              stroke-width="0"
              fill="#fff"
            />
          </transition>
          <path
            d="M137.45,134.74a6.9,6.9,0,0,1-4.61-2.09l-45-44c-2.28-2.23-2.92-5.46-1.38-7.06a3.28,3.28,0,0,1,2.39-.91,6.9,6.9,0,0,1,4.61,2.09l45.05,44c2.28,2.23,2.91,5.46,1.38,7.05A3.26,3.26,0,0,1,137.45,134.74Z"
          />
          <path
            d="M88.81,82.65A4.89,4.89,0,0,1,92,84.17l45.05,44c1.68,1.64,1.91,3.63,1.33,4.24a1.35,1.35,0,0,1-1,.29,4.89,4.89,0,0,1-3.21-1.51l-45-44c-1.68-1.64-1.92-3.64-1.34-4.24a1.34,1.34,0,0,1,1-.3m0-4A5.19,5.19,0,0,0,85,80.18h0c-2.33,2.41-1.69,6.83,1.42,9.87l45,44a8.82,8.82,0,0,0,6,2.65,5.21,5.21,0,0,0,3.84-1.52h0c2.32-2.41,1.68-6.83-1.43-9.87l-45-44a8.81,8.81,0,0,0-6-2.66Z"
          />
          <path
            d="M19.92,53.66c.84,50.11,74.87,50.12,75.72,0C94.8,3.54,20.77,3.54,19.92,53.66Z"
            fill="#c9fbff"
            stroke="#000"
            stroke-miterlimit="10"
            stroke-width="6"
          /></svg
      ></v-btn>
    </template>
    <span>Compare Meteorites</span>
  </v-tooltip>
</template>

<script>
export default {
  props: {
    compare: {
      type: Boolean,
      default: false
    }
  },
  data: () => ({
    height: "2em",
    width: "2em",
    stroke: 2
  }),
  methods: {
    compareMeteorite() {
      this.$emit("compareMeteorite");
    }
  }
};
</script>

<style>
#border {
  animation-name: bookmark-complete;
  animation-duration: 1s;
  animation-iteration-count: 1;
}
.border-enter {
  opacity: 0;
  stroke-dasharray: 0;
}

.border-enter-active,
.border-leave-active {
  transition: all 0.5s ease-in-out;
}

.border-enter-from,
.border-leave-to {
  opacity: 0;
  stroke-dasharray: 0;
}
</style>
