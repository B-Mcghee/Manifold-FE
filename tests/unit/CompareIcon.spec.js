import {  createLocalVue, shallowMount, mount } from '@vue/test-utils';
import Vue from 'vue'
import Vuetify from 'vuetify'
import CompareIcon from '@/components/UI/CompareIcon.vue'





describe('CompareIcon', () => {
  const localVue = createLocalVue();
  let vuetify;

  beforeEach(() => {
    vuetify = new Vuetify();

  })
  test('renders item', () =>{
    const wrapper = mount(CompareIcon,{
      localVue,
      vuetify, 
      propsData:{
        favorite:true
      }
    })
    expect(wrapper.html()).toMatchSnapshot();

  })

  test('button is firing and emitting event', () => {
    const wrapper = mount(CompareIcon, {
      localVue,
      vuetify, 
      propsData:{
        favorite:true
      },


    })
    expect(wrapper.find('.v-btn').exists()).toBe(true);

    const event = jest.fn();
    const button = wrapper.find('.v-btn');

    wrapper.vm.$on('compareMeteorite', event)

    expect(event).toHaveBeenCalledTimes(0)
    button.trigger('click')

    expect(event).toHaveBeenCalledTimes(1)
  })



})
