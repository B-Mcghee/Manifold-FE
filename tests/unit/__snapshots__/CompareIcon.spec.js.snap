// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompareIcon renders item 1`] = `<span class="v-tooltip v-tooltip--bottom" favorite="true"><!----><button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default" style="width: 100%;" aria-haspopup="true" aria-expanded="false"><span class="v-btn__content"><svg id="Layer_1" data-name="Layer 1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 163 158.49" height="2em" width="2em"><transition-stub name="border"><rect x="1" y="1" width="161" height="156.49" stroke="#000" stroke-miterlimit="10" stroke-width="0" fill="#fff"></rect></transition-stub> <path d="M137.45,134.74a6.9,6.9,0,0,1-4.61-2.09l-45-44c-2.28-2.23-2.92-5.46-1.38-7.06a3.28,3.28,0,0,1,2.39-.91,6.9,6.9,0,0,1,4.61,2.09l45.05,44c2.28,2.23,2.91,5.46,1.38,7.05A3.26,3.26,0,0,1,137.45,134.74Z"></path> <path d="M88.81,82.65A4.89,4.89,0,0,1,92,84.17l45.05,44c1.68,1.64,1.91,3.63,1.33,4.24a1.35,1.35,0,0,1-1,.29,4.89,4.89,0,0,1-3.21-1.51l-45-44c-1.68-1.64-1.92-3.64-1.34-4.24a1.34,1.34,0,0,1,1-.3m0-4A5.19,5.19,0,0,0,85,80.18h0c-2.33,2.41-1.69,6.83,1.42,9.87l45,44a8.82,8.82,0,0,0,6,2.65,5.21,5.21,0,0,0,3.84-1.52h0c2.32-2.41,1.68-6.83-1.43-9.87l-45-44a8.81,8.81,0,0,0-6-2.66Z"></path> <path d="M19.92,53.66c.84,50.11,74.87,50.12,75.72,0C94.8,3.54,20.77,3.54,19.92,53.66Z" fill="#c9fbff" stroke="#000" stroke-miterlimit="10" stroke-width="6"></path></svg></span></button></span>`;
